### ISSUE TITLE
Support UNIX timestamps (INTEGER) columns for SQL datetime field filter variables

### ISSUE THREAD

**Issue by ralf-rf** [metabase/metabase#4058]
Browser:  55.0.2883.95 (64-bit)
OS: MacOS
Database: BigQuery
Metabase Version: v0.22.0
Hosted: Google Compute Engine
Internal DB: H2

I have created a SQL based question which uses the following structure:

```sql
SELECT [stats.app_events_recent_page.page_id] AS page_id, total as count, ratio * 100 AS percent FROM (
 SELECT [stats.app_events_recent_page.page_id], count(*) AS total, RATIO_TO_REPORT(total) OVER() AS ratio
 FROM [stats.app_events_recent_page]
 WHERE [[{{datetime}} AND ]] [stats.app_events_recent_page.event_name] = 'page_view'
   AND [stats.app_events_recent_page.page_id] > 0 
   [[AND retailer_name = {{kies_een_retailer}}]]
   [[AND folder_version_id = {{folder_version_id}}]]
 GROUP by [stats.app_events_recent_page.page_id]
 ORDER BY [stats.app_events_recent_page.page_id] ASC
) 
```

I have created a Variable 'datetime' using a Field filter called TimestampMilliSeconds. The value of this field is in Milliseconds. The schema of the used table is also setup to treat the timestamp as an MS formatted one. The resulting query is:

```sql
SELECT [stats.app_events_recent_page.page_id] AS page_id, total as count, ratio * 100 AS percent FROM (
 SELECT [stats.app_events_recent_page.page_id], count(*) AS total, RATIO_TO_REPORT(total) OVER() AS ratio
 FROM [stats.app_events_recent_page]
 WHERE CAST(date([stats.app_events_recent_page.timestamp_micros]) AS timestamp) BETWEEN CAST('2016-12-28 00:00' AS timestamp) AND CAST('2017-01-03 00:00' AS timestamp) 
    AND  [stats.app_events_recent_page.event_name] = 'page_view'
    AND [stats.app_events_recent_page.page_id] > 0 
 GROUP by [stats.app_events_recent_page.page_id]
 ORDER BY [stats.app_events_recent_page.page_id] ASC
) 
```

Resulting in no results... because the timestamp is not converted to seconds first; but the MS are interpreteted as seconds instead.

The non custom SQL queries work as expected:
```sql
SELECT count(*) AS [count] 
FROM [stats.app_events_recent] 
WHERE ([stats.app_events_recent.event_name] = 'folder_view' 
  AND CAST(date(msec_to_timestamp([stats.app_events_recent.timestamp_micros])) AS timestamp) BETWEEN CAST(date(timestamp('2016-12-27T11:18:55.094Z')) AS timestamp) 
  AND CAST(date(timestamp('2017-01-03T11:18:55.094Z')) AS timestamp))
````

The **msec_to_timestamp** part is missing in the result of custom SQL query. Looks like a small bug; but would be very nice if you could clean this one!

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by AshrafSam** [comment]
@camsaul any workarounds for the time being?

**Comment by camsaul** [comment]
@AshrafSam unfortunately you'll have continuing doing things like 

```sql
WHERE date >= unix_timestamp({{date1}}) and date <= unix_timestamp({{date2}})
```

for the time being

**Comment by AshrafSam** [comment]
oh, i didn't actually want to use it this way, maybe i'll create views in
the database instead :(

**Comment by justathoughtor2** [comment]
Just wanted to throw in the two cents that I'm experiencing this issue as well with PostgreSQL and Unix Timestamp (Seconds) field types. Let me know if you need anyone to go into the code and see about a pull request. I'd be happy to do so, but just want to make sure I'm not duplicating the efforts of someone else, here.

Thanks!

**Comment by phpcodecrafting** [comment]
How is this still open?  Timestamp fields work with date filters when using the question builder, but are broken when writing via custom sql. If it's working correctly for question builder, seems like it shouldn't be too hard to also get it to work for custom sql? This is a massive and major issue for anyone who uses applications that use timestamps instead of datetime fields. This has been reported well over 1 year ago.

Can we please get an update on this?

**Comment by phpcodecrafting** [comment]
And can we please get this relabeled as a bug and not a proposal? #5071 is the same issue and got the label of bug and Priority 3.

**Comment by eschultze** [comment]
Found a workaround for sqlite if anyone is wondering
`[[ AND datetime(timestamp/1000, 'unixepoch') >= {{Start}} ]] [[ AND datetime(timestamp/1000, 'unixepoch') <= {{End}} ]]`

**Comment by liveayon** [comment]
Does anyone know how to fix it?

**Comment by maxgashkov** [comment]
Hi! Is there any way to speed up the resolution on this?

**Comment by ohld** [comment]
That feature would be really awesome!

**Comment by flamber** [comment]
This was fixed in 0.36.8
